### DISCUSSION TITLE
Secure upload to S3 cloudinary

### DISCUSSION THREAD

**Discussion by TomGriffiths88** [vercel/next.js#16509] [Help] [answered]
Hey guys,

I'm quite a new Dev and I'm starting to familiarise myself with Next to build my apps.

I want to upload images to S3 or cloudinary from an app and I'm getting a bit confused as to the best way to go about it. I know that usually image uploads are handled on the server for security reasons. And I definitely don't want to expose the credentials on the front end. But what's the best way to achieve this on next? 

If I store my sensitive credentials in a .env file. 
Should I secure an API end point with auth and then post from there? Or will that still expose the information on the front end? 

Any help or hints much appreciated

Cheers 
Tom

**Comment by leerob** [comment, marked as answer]
Here's an example for AWS S3!

https://github.com/leerob/nextjs-aws-s3/
